packages/customize-widgets/src/filters/move-to-sidebar.js

```diff
@@ -55,10 +55,10 @@ const withMoveToSidebarToolbarItem = createHigherOrderComponent(
 			const newSidebarControl = sidebarControls.find(
 				( sidebarControl ) => sidebarControl.id === sidebarControlId
 			);
-
-			if ( widgetId ) {
+			const activeSidebarAdapter = activeSidebarControl.sidebarAdapter;
+			if ( widgetId && activeSidebarAdapter.getWidget( widgetId ) ) {
```

Comment:
What's this change do?

Reply:
This was necessary to [fix a regression](https://github.com/WordPress/gutenberg/pull/34750/commits/766f0e6b22eae9fab28cd46bee9f653ea768043d).

When you move a block in the customizer to a different widget sidebar, previously we would check to see if the block has a widgetId and if so, assume it's a valid widget that we can just remove from the original sidebar and add to the destination sidebar. We would detect things like inner blocks by the absence of a widgetId (necessary in order to handle inner blocks differently.)

With the new implementation this can break when you take an existing widget and turn it into inner blocks by transforming it into a Group. The inner block retains its widgetId through this transformation, but the corresponding widget _instance_ is removed.

To cover this case, we now check both that the widgetId exists AND that there is an existing widget with that id.